modules/cloudflarepurger/src/Plugin/Purge/Purger/CloudFlarePurger.php

```diff
@@ -0,0 +1,210 @@
+   */
+  protected $state;
+
+  /**
+   * ZoneApi object for interfacing with CloudFlare Php Sdk.
+   *
+   * @var \CloudFlarePhpSdk\ApiEndpoints\ZoneApi
+   */
+  protected $zoneApi;
+
+  /**
+   * The current cloudflare ZoneId.
+   *
+   * @var string
+   */
+  protected $zone;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('cloudflare.state'),
+      $container->get('logger.factory')->get('cloudflare')
+    );
+  }
+
+  /**
+   * Constructs a \Drupal\Component\Plugin\CloudFlareTagPurger.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config
+   *   The factory for configuration objects.
+   * @param \Drupal\cloudflare\CloudFlareStateInterface $state
+   *   Tracks rate limits associated with CloudFlare Api.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   A logger instance.
+   *
+   * @throws \LogicException
+   *   Thrown if $configuration['id'] is missing, see Purger\Service::createId.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config, CloudFlareStateInterface $state, LoggerInterface $logger) {
+    $this->config = $config->get('cloudflare.settings');
+    $this->state = $state;
+    $this->logger = $logger;
+
+    // This is a unique case where the ApiSdk is being accessed directly and not
+    // via a service.  Purging should only ever happen through the purge module
+    // which is why this is NOT in a service.
+    $api_key = $this->config->get('apikey');
+    $email = $this->config->get('email');
+    $this->zone = $this->config->get('zone');
+    $this->zoneApi = new ZoneApi($api_key, $email);
+
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function invalidate(InvalidationInterface $invalidation) {
+    $this->invalidateMultiple([$invalidation]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function invalidateMultiple(array $invalidations) {
+    // @todo figure out why this is happening.
+    if (count($invalidations) == 0) {
+      $this->logger->warning("No Invalidations?  That's odd." . print_r($invalidations, TRUE));
+      return;
+    }
+
+    $chunks = array_chunk($invalidations, CloudFlareAPI::MAX_TAG_PURGES_PER_REQUEST);
+    foreach ($chunks as $chunk) {
+      $this->purgeChunk($chunk);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTimeHint() {
+    return 1.3;
```

Comment:
Are you also enforcing a outbound HTTP timeout? Else real-life code could still exceed that value and that would render the capacity API semi-useless :).